test/CellarWithMultiAssetDeposit.t.sol

```diff
@@ -0,0 +1,331 @@
+        // mockUsdtUsd.setMockAnswer(1e8);
+
+        // Setup pricing
+        PriceRouter.ChainlinkDerivativeStorage memory stor;
+        PriceRouter.AssetSettings memory settings;
+        uint256 price = uint256(mockUsdcUsd.latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, address(mockUsdcUsd));
+        priceRouter.addAsset(USDC, settings, abi.encode(stor), price);
+        price = uint256(mockWethUsd.latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, address(mockWethUsd));
+        priceRouter.addAsset(WETH, settings, abi.encode(stor), price);
+        price = uint256(mockWbtcUsd.latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, address(mockWbtcUsd));
+        priceRouter.addAsset(WBTC, settings, abi.encode(stor), price);
+        price = uint256(mockUsdtUsd.latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, address(mockUsdtUsd));
+        priceRouter.addAsset(USDT, settings, abi.encode(stor), price);
+
+        // Add adaptors and ERC20 positions to the registry.
+        registry.trustPosition(usdcPosition, address(erc20Adaptor), abi.encode(USDC));
+        registry.trustPosition(wethPosition, address(erc20Adaptor), abi.encode(WETH));
+        registry.trustPosition(wbtcPosition, address(erc20Adaptor), abi.encode(WBTC));
+        registry.trustPosition(usdtPosition, address(erc20Adaptor), abi.encode(USDT));
+
+        string memory cellarName = "Cellar V0.0";
+        uint256 initialDeposit = 1e6;
+        deal(address(USDC), address(this), initialDeposit);
+        USDC.approve(0xa0Cb889707d426A7A386870A03bc70d1b0697598, initialDeposit);
+        uint64 platformCut = 0.75e18;
+        cellar = new CellarWithMultiAssetDeposit(
+            address(this),
+            registry,
+            USDC,
+            cellarName,
+            "POG",
+            usdcPosition,
+            abi.encode(true),
+            initialDeposit,
+            platformCut,
+            type(uint192).max
+        );
+
+        // Set up remaining cellar positions.
+        cellar.addPositionToCatalogue(wethPosition);
+        cellar.addPosition(0, wethPosition, abi.encode(true), false);
+        cellar.addPositionToCatalogue(wbtcPosition);
+        cellar.addPositionToCatalogue(usdtPosition);
+        cellar.addPosition(0, usdtPosition, abi.encode(true), false);
+
+        cellar.setStrategistPayoutAddress(strategist);
+        vm.label(address(cellar), "cellar");
+        vm.label(strategist, "strategist");
+        // Approve cellar to spend all assets.
+        USDC.approve(address(cellar), type(uint256).max);
+        initialAssets = cellar.totalAssets();
+        initialShares = cellar.totalSupply();
+    }
+
+    // ========================================= HAPPY PATH TEST =========================================
+
+    // Can we accept the Curve LP tokens? I dont see why not.
```

Comment:
Is this note needed still? I think we move it to an issue if we are considering adding in Curve LP tokens and testing with them. Unless we think we don't need to test with them in an extra way.

Reply:
onlyOwner was made an internal function to save on contract size. When you use modifiers the modifier code is copied to all functions it decorates. By using an internal function modifier code is only copied once.

automation and viewPositionBalances were removed because they were not being used, and we needed more contract space. Thinking about it, being able to deposit multiple assets is much more valuable than automation actions.